linkml/generators/yumlgen.py

```diff
@@ -274,7 +298,13 @@ def prop_modifier(self, cls: ClassDefinition, slot: SlotDefinition) -> str:
     help="Name of the diagram in the output directory (without suffix!)",
 )
 def cli(yamlfile, **args):
-    """Generate a UML representation of a LinkML model"""
+    """Generate a UML representation of a LinkML model
+
+    .. warning::
+        `gen-yuml` is deprecated. Please use `gen-doc`, `gen-plantuml`, `gen-mermaid-class-diagram`, or `gen-erdiagram` instead.
```

Comment:
```suggestion
        `gen-yuml` is deprecated. Please use `gen-doc`, `gen-plantuml` or `gen-mermaid-class-diagram`.
```

yuml generator generates a yUML class diagram, so the alternative class diagram generators would be just the above three.